llama_stack/providers/inline/agents/meta_reference/agent_instance.py

```diff
@@ -156,7 +156,7 @@ async def create_and_execute_turn(
         turns = await self.storage.get_session_turns(request.session_id)
 
         messages = []
-        if len(turns) == 0 and self.agent_config.instructions != "":
+        if self.agent_config.instructions != "":
```

Comment:
sorry this doesn't seem right. because we do hydrate messages from the previous turns and those would have a SystemMessage right?

Reply:
Thats the problem, I printed messages that are being sent into inference, and also printed the turns that are being fetched from persistence store and after the first turn, they do not have the SystemMessage in them which causes the agent flow to go off track
Thats why I made this change to make sure messages always has a system message in it

Reply:
```
class Turn(BaseModel):
    """A single turn in an interaction with an Agentic System."""

    turn_id: str
    session_id: str
    input_messages: List[
        Union[
            UserMessage,
            ToolResponseMessage,
        ]
    ]
    steps: List[Step]
    output_message: CompletionMessage
    output_attachments: List[Attachment] = Field(default_factory=list)

    started_at: datetime
    completed_at: Optional[datetime] = None
```
This is the Turn pydantic Object and input messages are of type UserMessage or ToolResponseMessage and output_message is of type CompletionMessage.
@ashwinb

Reply:
@cheesecake100201 you are right - we don't store this in the Turn. this fix is good!